bundles/org.openhab.core.config.discovery/src/main/java/org/openhab/core/config/discovery/internal/PersistentInbox.java

```diff
@@ -192,10 +196,10 @@ protected void deactivate() {
         ThingUID newThingUID = thingUID;
         if (newThingId != null) {
             try {
-                newThingUID = new ThingUID(thingTypeUID, newThingId);
+                newThingUID = new ThingUID(thingUID.getAsString().substring(0,
+                        thingUID.getAsString().lastIndexOf(AbstractUID.SEPARATOR) + 1) + newThingId);
             } catch (IllegalArgumentException e) {
-                logger.warn("Cannot create thing: {}", e.getMessage());
-                return null;
+                throw new IllegalArgumentException("Cannot create thing UID: " + e.getMessage());
```

Comment:
This could just rethrow the original exception.
Or at least it could use the original exception as parent.

Reply:
Changed